transcom/mymove
# INT B-17272, B-19828, B-19829 backend okta response check customer app

> [!NOTE]  
> This PR is the cumulation of three separate backlog items.
## [Agility ticket - B-17272](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-17272)
## [Agility ticket - B-19828](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19828)
## [Agility ticket - B-19829](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19829)

## Summary

In the customer app, when a user attempts to authenticate with Okta with the same DODID/email as an office user they are redirected back to MilMove due to configuration rules we have set in the Okta Admin Console.

This PR is work for all of the above backlog items and each one achieves the following:
> **B-17272 Okta Config: Add redirect w/params to redirect uris in Okta Customer app (test & prod)**

- Before we can handle redirects, we must allow the redirect uri in Okta configs for each application. This was completed and updated in the Okta Admin Console and this can be "tested" by ensuring we don't get an Okta error page when being redirected back.

> **B-19828 Backend: Check for Okta "access denied" response when authenticating and add redirect w/parameters**
- we were already checking for the `access_denied` error description in the backend and to avoid confusion, I adjusted the redirect params that are appended to be more generalized when the user is denied access from Okta
- this is because Okta doesn't specify WHY the user was denied access (lame)
- updated tests

> **B-19829 UI Customer: Set up params and display banner when Customer is redirected from Okta with "access denied" error**
- added `OktaErrorBanner` component that handles what displays in the banner, added storybook & test
- updated existing param check in the office application to consider `okta_error` param instead of `okta_logged_out` to adjust to changed backend redirect
- added param check & state change to customer application (similar to what's being done in the office app) and leverage state to conditionally render the new banner

> [!NOTE]  
> This also contains an update to the `run-server-test` script that checks for operating system to get rid of some annoying text when running `make server_test` <br> <br>as well as increasing the timeout in `ServicesCounselingEditShipmentDetails.test.jsx`


### How to test

1. Ensure that you are able to access admin/office applications through Okta
2. If you aren't sure, contact an Okta admin (myself, Dre, Cameron) to add you to the respective groups in the Okta Admin Console
3. Whip up the customer app after firing off your `server` and `client`
4. Sign in, Accept Terms, and try to sign in with your CAC into the customer application
5. You should be redirected back and see a banner


https://github.com/transcom/mymove/assets/136510600/9b9f366b-e725-4692-a859-d25c60ec7859

## Screenshots
![Screenshot 2024-06-11 at 10 02 28 AM](https://github.com/transcom/mymove/assets/136510600/cf282608-396e-485a-b396-32baef20bc2c)
